iroh-net/src/discovery/dns.rs

```diff
@@ -36,12 +42,41 @@ const DNS_STAGGERING_MS: &[u64] = &[200, 300];
 #[derive(Debug)]
 pub struct DnsDiscovery {
     origin_domain: String,
+    port: Option<u16>,
+    use_ipv6: bool,
 }
 
 impl DnsDiscovery {
     /// Create a new DNS discovery.
     pub fn new(origin_domain: String) -> Self {
-        Self { origin_domain }
+        Self {
+            origin_domain,
+            port: None,
+            use_ipv6: false,
+        }
+    }
+
+    /// Create a new DNS discovery with a custom port and IP protocol version.
```

Comment:
```suggestion
    /// Creates a new DNS discovery with a custom port and IP protocol version.
```

We should try and follow https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text for doc comments (though we're not consistently there yet, so there are plenty of bad examples)

Reply:
I'm indeed not good at documenting comments, and my English is not very good either; I will learn more about both.

Reply:
Thank you, I think your comments are very well written. I will carefully read the content of this link to improve myself